lib/ansible/module_utils/oneview.py

```diff
@@ -0,0 +1,782 @@
+
+        for i, val in enumerate(resource1):
+            if isinstance(val, dict):
+                # change comparison function to compare dictionaries
+                if not ResourceComparator.compare(val, resource2[i]):
+                    logger.debug("resources are different. " + debug_resources)
+                    return False
+            elif isinstance(val, list):
+                # recursive call
+                if not ResourceComparator.compare_list(val, resource2[i]):
+                    logger.debug("lists are different. " + debug_resources)
+                    return False
+            elif ResourceComparator._standardize_value(val) != ResourceComparator._standardize_value(resource2[i]):
+                logger.debug("values are different. " + debug_resources)
+                return False
+
+        # no differences found
+        return True
+
+    @staticmethod
+    def _str_sorted(obj):
+        if isinstance(obj, dict):
+            return json.dumps(obj, sort_keys=True)
+        else:
+            return str(obj)
+
+    @staticmethod
+    def _standardize_value(value):
+        """
+        Convert value to string to enhance the comparison.
+
+        Args:
+            value: Any object type.
+
+        Returns:
+            str: Converted value.
+        """
+        if isinstance(value, float) and value.is_integer():
+            # Workaround to avoid erroneous comparison between int and float
+            # Removes zero from integer floats
+            value = int(value)
+
+        return str(value)
+
+
+class ResourceMerger():
+    @staticmethod
+    def merge_list_by_key(original_list, updated_list, key, ignore_when_null=[]):
+        """
+        Merge two lists by the key. It basically:
+        1. Adds the items that are present on updated_list and are absent on original_list.
+        2. Removes items that are absent on updated_list and are present on original_list.
+        3. For all items that are in both lists, overwrites the values from the original item by the updated item.
+
+        Args:
+            original_list: original list.
+            updated_list: list with changes.
+            key: unique identifier.
+            ignore_when_null: list with the keys from the updated items that should be ignored in the merge, if its
+            values are null.
+        Returns:
+            list: Lists merged.
+        """
+        if not original_list:
+            return updated_list
+
+        items_map = collections.OrderedDict([(i[key], i.copy()) for i in original_list])
+
+        merged_items = collections.OrderedDict()
+
+        for item in updated_list:
+            item_key = item[key]
+            if item_key in items_map:
+                for ignored_key in ignore_when_null:
+                    if ignored_key in item and not item[ignored_key]:
+                        item.pop(ignored_key)
+                merged_items[item_key] = items_map[item_key].copy()
```

Comment:
This will fail if ```updated_list``` has items which are not present in ```original_list```.  You probably want something like this instead:
``` python
merged_items[item_key] = items_map.get(item_key, {})
merged_items[item_key].update(item)
```